tools/windows/DatadogAgentInstaller/CustomActions/UserCustomActions.cs

```diff
@@ -47,6 +47,95 @@ private static string GetDefaultDomainPart()
             return Environment.MachineName;
         }
 
+        private static ActionResult ReadRegistryProperties(ISession session)
+        {
+            // This CA runs (only once) in either the InstallUISequence or the InstallExecuteSequence.
+            // For UI installs it ensures that the agent user dialog is pre-populated with registry
+            // creds if applicable.
+            try
+            {
+                // DDAGENTUSER_NAME
+                //
+                // The user account can be provided to the installer by
+                // * The registry
+                // * The command line
+                // * The agent user dialog
+                // The user account domain and name are stored separetely in the registry
+                // but are passed together on the command line and the agent user dialog.
+                // This function will combine the registry properties if they exist.
+                // Preference is given to creds provided on the command line and the agent user dialog.
+                if (string.IsNullOrEmpty(session["DDAGENTUSER_NAME"]))
+                {
+                    try
+                    {
+                        var domain = Microsoft.Win32.Registry.LocalMachine.OpenSubKey(@"Software\Datadog\Datadog Agent")
+                            .GetValue("installedDomain")
+                            .ToString();
+                        var user = Microsoft.Win32.Registry.LocalMachine.OpenSubKey(@"Software\Datadog\Datadog Agent")
+                            .GetValue("installedUser")
+                            .ToString();
```

Comment:
Seems to better to open subkey once. Also I have seen a pattern for this using "using" which will force the closure of the key on exit.